noir-projects/aztec-nr/aztec/src/note/note_getter.nr

```diff
@@ -121,6 +123,13 @@ pub fn get_notes<Note, N, FILTER_ARGS>(
             context.push_note_hash_read_request(note_hash_for_read_request);
 
             num_notes += 1;
+
+            if (write_index != i) {
+                opt_notes[write_index] = Option::some(note);
+                opt_notes[i] = Option::none();
+            }
```

Comment:
Could you document here what this is doing? I would be very confused what's going on if I didn't have the context.